frappe/utils/bench_helper.py

```diff
@@ -66,7 +54,45 @@ def get_sites(site_arg: str) -> list[str]:
 			fg="red",
 		)
 
-	return []
+	if site is not None:
+		try:
+			frappe.init(site=site)
+		except frappe.exceptions.IncorrectSitePath:
+			click.secho(f"Site {site} does not exist!", fg="yellow")
+			ctx.exit(1)
+		ctx.default_map = frappe.conf
```

Comment:
The potential new defaults, mapping options to `frappe.conf` keys.

At the moment, this PR only anticipates `frappe.init` and deals with the direct consequences of this, by adopting `with_site` and `with_each_site` context managers.